lib/Parse/DNS/Zone.pm

```diff
@@ -408,6 +413,42 @@ sub _load_zonefile {
 	return do { local $/; <$zonefh> }; # slurp
 }
 
+sub _parse_time_unit {
+	my $string = shift;
+
+	my $time_units_regex = qr/^
+			(?: (?=\d+(?: w|d|h|m|s))
+			(\d+w)? (\d+d)? (\d+h)? (\d+m)? (\d+s)? ) | (\d+)
+		$/xi;
```

Comment:
This is over-indented, the final `$/xi;` should line up with `my` (and the two lines above should as such be deintended one level. My patch also made a simplification where we avoid having `$string` and `$time_units_regexp`, and instead just applied the regexp directly to shift.